keras_cv/models/resnet_v2.py

```diff
@@ -325,90 +429,144 @@ def ResNetV2(
+                stride=stackwise_strides[stack_index],
+                dilation=stackwise_dilations[stack_index],
+                block_fn=block_fn,
+                first_shortcut=block_fn == Block or stack_index > 0,
+                stack_index=stack_index,
+            )(x)
+            stack_level_outputs[stack_index + 2] = x
+
+        x = layers.BatchNormalization(axis=BN_AXIS, epsilon=BN_EPSILON, name="post_bn")(
+            x
+        )
+        x = layers.Activation("relu", name="post_relu")(x)
 
-    return model
+        if include_top:
+            x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+            x = layers.Dense(
+                classes, activation=classifier_activation, name="predictions"
+            )(x)
+        else:
+            if pooling == "avg":
+                x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+            elif pooling == "max":
+                x = layers.GlobalMaxPooling2D(name="max_pool")(x)
+
+        super().__init__(inputs=inputs, outputs=x, **kwargs)
+
+        # All references to `self` below this line.
+        if weights is not None:
+            self.load_weights(weights)
+        # Set this private attribute for recreate backbone model with outputs at each of the
+        # resolution level.
+        self._backbone_level_outputs = stack_level_outputs
+
+        # Bind the `to_backbone_model` method to the application model.
+        self.as_backbone = types.MethodType(utils.as_backbone, self)
+
+        self.stackwise_filters = stackwise_filters
+        self.stackwise_blocks = stackwise_blocks
+        self.stackwise_strides = stackwise_strides
+        self.include_rescaling = include_rescaling
+        self.include_top = include_top
+        self.stackwise_dilations = stackwise_dilations
+        self.input_tensor = input_tensor
+        self.pooling = pooling
+        self.classes = classes
+        self.classifier_activation = classifier_activation
+        self.block_fn = block_fn
+
+    def get_config(self):
+        return {
+            "stackwise_filters": self.stackwise_filters,
+            "stackwise_blocks": self.stackwise_blocks,
+            "stackwise_strides": self.stackwise_strides,
+            "include_rescaling": self.include_rescaling,
+            "include_top": self.include_top,
+            "input_shape": self.input_shape[1:],
+            "stackwise_dilations": self.stackwise_dilations,
+            "input_tensor": self.input_tensor,
+            "pooling": self.pooling,
+            "classes": self.classes,
+            "classifier_activation": self.classifier_activation,
+            "block_fn": keras.utils.get_registered_name(self.block_fn),
+            "name": self.name,
+            "trainable": self.trainable,
+        }
+
+    @classmethod
+    def from_config(cls, config):
+        block_class_name = config["block_fn"]
+        config["block_fn"] = keras.utils.get_registered_object(block_class_name)
```

Comment:
nice - glad to see there's a way to do this.  I have actually not seen these utilities before.